pkg/segment/reader/segread/segreader/segreader.go

```diff
@@ -169,13 +247,21 @@ func (sfr *SegmentFileReader) Close() error {
 	if sfr.currFD == nil {
 		return errors.New("SegmentFileReader.Close: tried to close an unopened segment file reader")
 	}
-	sfr.ReturnBuffers()
+	err := sfr.ReturnBuffers()
+	if err != nil {
+		return err
+	}
 	return sfr.currFD.Close()
 }
 
-func (sfr *SegmentFileReader) ReturnBuffers() {
-	UncompressedReadBufferPool.Put(&sfr.currRawBlockBuffer)
+func (sfr *SegmentFileReader) ReturnBuffers() error {
+	// UncompressedReadBufferPool.Put(&sfr.currRawBlockBuffer)
 	FileReadBufferPool.Put(&sfr.currFileBuffer)
+	err := PutBufToPool(sfr.currRawBlockBuffer)
+	if err != nil {
+		return fmt.Errorf("Error putting Raw Block Buffer back to pool")
```

Comment:
nit: instead of `fmt.Errorf("Error putting Raw Block Buffer back to pool")` I would do 
`fmt.Errorf("ReturnBuffers: Error putting Raw Block Buffer back to pool, err: %v", err)`